src/axom/primal/tests/primal_bezier_curve.cpp

```diff
@@ -118,64 +118,161 @@ TEST(primal_beziercurve, evaluate)
   using PointType = primal::Point<CoordType, DIM>;
   using BezierCurveType = primal::BezierCurve<CoordType, DIM>;
 
-  const int order = 3;
-  PointType data[order + 1] = {PointType {0.6, 1.2, 1.0},
-                               PointType {1.3, 1.6, 1.8},
-                               PointType {2.9, 2.4, 2.3},
-                               PointType {3.2, 3.5, 3.0}};
+  const int max_order = 3;
+  PointType data[max_order + 1] = {PointType {0.6, 1.2, 1.0},
+                                   PointType {1.3, 1.6, 1.8},
+                                   PointType {2.9, 2.4, 2.3},
+                                   PointType {3.2, 3.5, 3.0}};
 
-  BezierCurveType b2Curve(data, order);
+  // clang-format off
+  PointType exp_vals[4][3] = {{PointType {0.6, 1.2, 1.0}, PointType {  0.6,    1.2,    1.0}, PointType {0.6, 1.2, 1.0}},
+                              {PointType {0.6, 1.2, 1.0}, PointType { 0.95,    1.4,    1.4}, PointType {1.3, 1.6, 1.8}},
+                              {PointType {0.6, 1.2, 1.0}, PointType {1.525,    1.7,  1.725}, PointType {2.9, 2.4, 2.3}},
+                              {PointType {0.6, 1.2, 1.0}, PointType { 2.05, 2.0875, 2.0375}, PointType {3.2, 3.5, 3.0}}};
+  // clang-format on
+
+  // Test evaluation at various orders, using the first `ord`
+  //  points in `data` as control points.
+  for(int ord = 0; ord <= max_order; ++ord)
+  {
+    BezierCurveType curve(data, ord);
 
-  PointType midtval {2.05, 2.0875, 2.0375};
+    PointType calc_start = curve.evaluate(0.0);
+    PointType calc_mid = curve.evaluate(0.5);
+    PointType calc_end = curve.evaluate(1.0);
 
-  // Evaluate the curve at several parameter values
-  // Curve should interpolate endpoints
-  PointType eval0 = b2Curve.evaluate(0.0);
-  PointType eval1 = b2Curve.evaluate(1.0);
-  PointType evalMid = b2Curve.evaluate(0.5);
+    for(int i = 0; i < DIM; ++i)
+    {
+      EXPECT_NEAR(calc_start[i], exp_vals[ord][0][i], 1e-15);
+      EXPECT_NEAR(calc_mid[i], exp_vals[ord][1][i], 1e-15);
+      EXPECT_NEAR(calc_end[i], exp_vals[ord][2][i], 1e-15);
+    }
+  }
+}
 
-  for(int i = 0; i < DIM; ++i)
+//------------------------------------------------------------------------------
+TEST(primal_beziercurve_, first_derivatives)
+{
+  SLIC_INFO("Testing Bezier derivative calculation");
+
+  const int DIM = 3;
+  using CoordType = double;
+  using PointType = primal::Point<CoordType, DIM>;
+  using VectorType = primal::Vector<CoordType, DIM>;
+  using BezierCurveType = primal::BezierCurve<CoordType, DIM>;
+
+  const int max_order = 3;
+  PointType data[max_order + 1] = {PointType {0.6, 1.2, 1.0},
+                                   PointType {1.3, 1.6, 1.8},
+                                   PointType {2.9, 2.4, 2.3},
+                                   PointType {3.2, 3.5, 3.0}};
```

Comment:
I'm a little surprised you don't need `clang-format off` before the declaration of `data`.  OK!